src/lib/mafia/roles/town/Crusader.ts

```diff
@@ -0,0 +1,73 @@
+import SingleTarget from '@mafia/mixins/SingleTarget';
+import Townie from '@mafia/mixins/Townie';
+import NightActionsManager, { Attack, NightActionPriority } from '@mafia/managers/NightActionsManager';
+import Player from '@mafia/structures/Player';
+
+class Crusader extends SingleTarget {
+
+	public name = 'Crusader';
+	public description = 'You may protect someone every night, attacking one randomly selected visitor if your target is visited.';
+	public action = 'protect';
+	public actionText = 'protect a player';
+	public actionGerund = 'protecting';
+	public priority = NightActionPriority.CRUSADER;
+	private isTargetAttacked = false;
+
+	public runAction(actions: NightActionsManager, target: Player) {
+		const playerRecord = actions.record.get(target.user.id);
+		if (playerRecord.size === 0) {
+			return;
+		}
+
+		// Select visitor to be killed.
+		let visitors: Set<Player> = new Set();
+		let actionNames = [...playerRecord.keys()];
```

Comment:
alternatively, just `const { visitors } = this.player`. For this to work though, you'll have to put Crusader's priority dead last, next to Ambusher.